docs/proposals/20200720-single-controller-multitenancy.md

```diff
@@ -0,0 +1,548 @@
+identity with RBAC rights to provision the infrastructure only in the Subscription. The workload 
+clusters must run with a System Assigned machine identity. The organization has adopted Cluster API 
+in order to manage Kubernetes infrastructure, and expects 'management' clusters running the Cluster 
+API controllers to manage 'workload' clusters in dedicated Azure Subscriptions with an AAD account 
+which only have access to that Subscription.
+
+The current configuration exists:
+* Subscriptions for each cluster
+* AAD Service Principals with Subscription Owner rights for each Subscription
+* A management Kubernetes cluster running Cluster API Provider Azure controllers
+
+Alex can provision a new workload cluster in the specified Subscription with the corresponding AAD 
+Service Principal by creating new Cluster API resources in the management cluster. Each of the
+workload cluster machines would run as the System Assigned identity described in the Cluster API
+resources. The CAPZ controller in the management cluster uses the Service Principal credentials when
+reconciling the AzureCluster so that it can create/use/destroy resources in the workload cluster.
+
+
+#### Story 2 - Legacy Behavior Preserved
+
+Dascha is an engineer in a smaller, less strict organization with a few Azure accounts intended to 
+build all infrastructure. There is a single Azure Subscription named 'dev', and Dascha wants to 
+provision a new cluster in this Subscription. An existing Kubernetes cluster is already running the 
+Cluster API operators and managing resources in the dev Subscription. Dascha can provision a new 
+cluster by creating Cluster API resources in the existing cluster, omitting the ProvisionerIdentity 
+field in the AzureCluster spec. The CAPZ operator will use the Azure credentials provided in its 
+deployment template.
+
+#### Story 3 - Software as a Service Provider
+
+ACME Industries is offering Kubernetes as a service to other organizations. ACME creates an AAD 
+Identity for each organization and each organization grants that Identity access to provision 
+infrastructure in one or multiple of their Azure Subscriptions. ACME Industries wants to minimise 
+the memory footprint of managing many clusters, and wants to move to having a single instance of 
+CAPZ to managed infrastructure across multiple organizations.
+
+## Requirements
+
+### Functional
+
+<a name="FR1">FR1.</a> CAPZ MUST support assuming an identity specified by an `AzureCluster.ProvisioningIdentity`.
+
+<a name="FR2">FR2.</a> CAPZ MUST support static credentials.
+
+<a name="FR3">FR3.</a> CAPZ MUST prevent privilege escalation allowing users to create clusters in Azure accounts they should
+  not be able to.
+
+<a name="FR4">FR4.</a> CAPZ SHOULD support credential refreshing modified principal data.
+
+<a name="FR5">FR5.</a> CAPZ SHOULD provide validation for principal data submitted by users.
```

Comment:
what kind of validation? Does that include validating that the credentials are valid/not expired?

Reply:
It would be nice to attempt to get a token from AAD. If we are unable to get a token from AAD, perhaps we get a 401, we should reject the secret as invalid.